vars/buildExecute.groovy

```diff
@@ -102,7 +105,14 @@ void call(Map parameters = [:]) {
                     error "[${STEP_NAME}] buildTool not set and no dockerImage & dockerCommand provided."
                 }
         }
-
+        if (config.cnbBuild) {
+            if (!['npm', 'gradle', 'maven', 'mta', 'docker'].contains(config.buildTool)) {
```

Comment:
Can you turn this condition around or use switch for better readability?

Reply:
Is it correct that `kaniko` is not supported?

Reply:
> Can you turn this condition around or use switch for better readability?

When I turned it around before, it broke the check. But I can look into the right way to format this.

Reply:
> Is it correct that `kaniko` is not supported?

They don't make sense together, as both kanikoExecute and cnbBuild are pushing images to a registry...

Reply:
> Is it correct that `kaniko` is not supported?

@pbusko @c0d1ngm0nk3y should we be supporting the option that both kanikoExecute and cnbBuild could be run? For example we could change it so that kaniko buildTool only ever runs kanikoExecute (and not cnbBuild) but that if someone were to set up the step kanikoExecute itself (regardless of buildTool) it would run. What do you think? What really makes sense here? I'm not the expert. I'm assuming they don't make sense together, but maybe they could.

Reply:
> @pbusko @c0d1ngm0nk3y should we be supporting the option that both kanikoExecute and cnbBuild could be run?

No, that should not be the case.